libbf-quickjs-2021-03-27/libbf.h

```diff
@@ -27,7 +27,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
-#if defined(__x86_64__)
+#if INTPTR_MAX >= INT64_MAX
```

Comment:
This is the most important change, as this in particular is responsible for the misbehavior on AArch64. (Everything else is just an added bonus.)

Reply:
Yes I tested this on my M2 Mac and the problem seems to be resolved.